### DISCUSSION TITLE
Access denied

### DISCUSSION THREAD

**Discussion by xSido** [boxbilling/boxbilling#1022] [Q&A] [answered]
First of all, hello guys, and thank you for all the efforts you are putting in.

I'm new to git, so please excuse me if I'm not posting in the right section.

I currently got a problem with version 4.2.2 beta 1
I.ve installed and configured it. Everything seems to work just fine, except when i want to delete an order.
It tells me : WHM server response error calling action removeacct: "Permission Denied"
Same for suspend (with suspend permission denied ofc)

And since I'm here, another thing I noticed is when u click to confirm the order as a client, if you double click it fast, it create it several orders with same specifications, only one being active and the rest waiting for activation.

Edit:
Because of the above problem, i got 3 orders.
1 - active and functional (original one)
2 - require activation - its not functional and is not made in whm
3 - exactly like 2
2 and 3 were made because i forced the order button\check out and requires activation because 1 is created and active with same information

Now getting back to accces denied. I cant delete none of these orders.
But i found a solution:
-for the bugged orders i have to delete them from database, and they disappear from client and admin panel.
-for the working one, i have to delete from both whm and databse.

Edit2:
From what I've read so far, i guess the problem is root privileges which i miss.
If that is the case. There might be a half solution to it. 
A sql command that will allow u remove order from db and then tell u to log whm to remove it from there manually as well.

**Comment by BelleNottelling** [comment]
On the subject of the "permission denied" message. you should check that the details you are using with whm has the permission to make those changes

**Reply by xSido** [reply]
Hey, thanks for answering.
To make it clear for me, does this require something i have to check in whm, in boxbilling settings or hosting provider?

**Reply by BelleNottelling** [reply]
I would assume whm, but I haven't used it. One of these links may help you. That server manager is also quite old, so it is possible it doesn't work correctly anymore.
Links:
https://docs.whmcs.com/Administrators_and_Permissions
or
https://docs.cpanel.net/manage2/users/add-user/

**Reply by xSido** [reply]
Well, I can't find any settings to fit those options. 
I guess it requires root settings which I don't have.


As i said in edit 2, i believe there is nothing i can do.
But, to make it easier for anyone using boxbilling and not having the access, it could be made an update to add an option to remove orders and any other data for which is required root access, from database. Because data displayed on site is stored in database. And after, promt u a message with what u have to remove manually from whm.

**Reply by BelleNottelling** [reply]
If you could create an issue or feature request for that part, I believe it's a worthwhile change to consider.
https://github.com/boxbilling/boxbilling/issues/new/choose

**Reply by xSido** [reply]
Thank you!

**Comment by xSido** [comment]
After discussing with @BenNottelling  i thought i need root access, so I tried to talk with my host provider but was a NO.

I quit looking for a solution but i had that feeling that i remember it worked. After all... that's why i made this post, cuz i remember it worked and somehow it didn't anymore.

And voila! I got in my application installer (Installatron in my case) the possibility to install BoxBilling, but version 3.6.11 or 2.12.5.
Right now i installed 3.6.11 version which require php 5.2 but work on 5.6 (cuz i don't have lower), and everything i stated above as not working, is working fine.

Maybe anyone got the experience to look into what is happening.
I believe it's either the version or the installatron which makes it work.

**Reply by BelleNottelling** [reply]
We may need to investigate the compatibility of the module with PHP 7.2+
Also, I highly discourage using such an old version of BB and PHP due to inherent security risks with running outdated software

**Reply by xSido** [reply]
I'm not intending to use this version. ATM i'm just testing/playing/learning it, but tried to give a hint to the problem i had/found with version 4.2.2 beta 1

**Comment by xSido** [comment]
i found a solution to this problem.
i used the WHM.php file from version 3.6.11 (which worked fine for me) to 4.2.2 beta 1 version, from this location directory/bb-library/Server/Manager/WHM.php and everything work fine.

**Reply by xSido** [reply]
i don't know what to believe anymore.
today i tried all day via phone to install BoxBilling and no matter what i've done, i still received access denied (one thing to mention, both version 3.6.11 and 4.2.2 i downloaded on phone had the same code inside whm.php, which yesterday was different)
now that i'm home, i tried again.... and now it works without changing the whm.php file........... WTF!?

installatron 3.6.11 work fine
boxbilling .org / .com download stable beta 1 version 4.2.2 work fine
updating 4.2.2 version with git version work fine

**Reply by yagiz-dev** [reply]
it's expected. the module itself has probably never been touched since years.

**Reply by BelleNottelling** [reply]
@xSido looking at the history of that module, it hasn't actually had a functional change since 2014, so there's no real reason for using an older version to work or behave any different from what we have now.
It ultimately needs to be rewritten

**Reply by xSido** [reply]
@BenNottelling  atm i'm running on latest files (updated from github) with no problem.

i really don't know what happened.

**Comment by xSido** [comment]
After many other tests. on all versions i could grab ( Installatron 3x , beta 4.x and github lattest version)
No matter the way i installed them. This is what i know so far.

-not having root access to my account is not the reason.
-editing configs or files is not the reason.
-does not matter if u install it on main reseller account or not.
-does not matter how u make the products/hosting plans.

i got on same install orders that i can suspend and orders that i can't.
i believe BoxBilling have a problem with creating orders and setting their owner rights where administrator is not seen as an owner/capable to edit that order.
or.... domain name (which affecting user/client etc might be the cause) -  for beta.com i get access domain no matter what.

**Comment by xSido** [comment, marked as answer]
After many other tests i found the problem.

In username column from service_hosting table must be used small characters.

When i ordered a host product, i wrote domain name in this format: Name.com instead of: name.com and this made the site create everything related to it, in database, with first letter BIG, including username. Whm, on the other side, use small letters and created the new order with small letter, therefore admin panel couldn't identify the user with BIG letters and gave the error access denied.

Editing the info in that column make everything work as normal.

**Reply by BelleNottelling** [reply]
It sounds like we just need to lowercase the input on that field, should be simple

**Reply by xSido** [reply]
Yeah, I'm trying to figure it out how to do it 👯‍♂️

**Reply by BelleNottelling** [reply]
> thank you for doing this...this has been on my list to do for awhile and i was actually just sitting down to download the latest and make the change and BOOM! there it is. now on to requiring phone number on new account registration....

Doing what, exactly?

**Reply by dbuerer** [reply]
@BenNottelling sorry i deleted my response...for some reason i responded to the incorrect posting
